### ISSUE TITLE
Highlight empty lines?

### ISSUE THREAD

**Issue by herbygillot** [dandavison/delta#137]
The standard pager for git will highlight additions of lines whose only content are spaces or tabs in red.  Is it possible to have similar behavior in delta?

Thanks in advance.

**Comment by dandavison** [comment]
Hi @herbygillot, this is a good call, thanks!

Notes to help implementing a fix / unit tests for this:

Here's an example

```
diff --git a/z.js b/z.js
index cf70fdb..c528683 100644
--- a/z.js
+++ b/z.js
@@ -3,5 +3,6 @@ function test() {
 }
 
 
+   
 test();
 console.log("this isn't in the hunk!")
```

which `git --no-pager diff` displays as 

<table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/80492773-7dca4700-8932-11ea-82f7-f1a6c3af3345.png" alt="image" /></td></tr></table>

but with delta this is 

<table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/80492860-989cbb80-8932-11ea-9793-c38edcbdfd9d.png" alt="image" /></td></tr></table>

**Comment by aureliojargas** [comment]
By reading the documentation, I was expecting that forcing something like `--whitespace-error-style 'red reverse'` would give me that behaviour.

> --whitespace-error-style <whitespace-error-style>
> Style for whitespace errors. Defaults to color.diff.whitespace if that is set in git config, or else
> 'magenta reverse' [default: auto auto]

Unfortunately, it also does not work.

**Comment by dandavison** [comment]
Hi @aureliojargas, let me give a quick response where I may be missing something:

I actually thought that this was fixed and that I'd forgotten to close the issue. Which delta version are you using? With 0.3.0 this is what I get using the diff pasted above. Is this what you want the output to be?

<table><tr><td><img width=600px src="https://user-images.githubusercontent.com/52205/88436508-6d5f2680-cdd2-11ea-9cca-fb7f4767e797.png" alt="image" /></td></tr></table>

**Comment by aureliojargas** [comment]
Hi @dandavison, thanks for the super quick answer!

From what I can tell, it seems it is working correctly for lines containing only whitespace (as in your test), but not for lines with content and trailing whitespace. I'm using delta 0.3.0 in macOS 10.13.6.

Considering a text file `foo` that has a single `x` line, I've added two spaces after the `x` and a new line with another two spaces.

```
echo x > foo
git add foo
echo -e "x  \n  "  > foo
git diff foo > foo.diff
```

Note that the plain git diff marks both cases as red, while delta does it only in the empty line:

![image](https://user-images.githubusercontent.com/282592/88439046-62c08380-ce0a-11ea-8889-6f36b6b3d23d.png)

Please tell me if you need additional tests or information from my side, I'm glad to help.

**Comment by dandavison** [comment]
Thanks @aureliojargas that’s very helpful. I should be able to get a fix in shortly.

**Comment by ryuheechul** [comment]
It's been a week that I've been enjoying delta and I just arrived to this issue too 😅  hence curious about updates on the issue as well.

**Comment by dvoytik** [comment]
Yeah, this is really missing feature from standard git diff output to highlight trailing white-spaces.
Please let me know if you need help and would accept a patch fixing this.

**Comment by dandavison** [comment]
@dvoytik sorry I missed your offer of help here. There's discussion in https://github.com/dandavison/delta/pull/1037 and @amtoine has offered to work on this. What I'm hoping to do is first get some tests that demonstrate exactly what we're trying to fix. Once we have those tests let's check that we're all in agreement about what the behavior should be.

**Comment by dandavison** [comment]
@dvoytik @aureliojargas @herbygillot if you are able to test the branch https://github.com/dandavison/delta/pull/1037 to see whether it behaves as you would want w.r.t. whitespace changes that would be fantastic. The work in that branch has been done by @wescande.

**Comment by aureliojargas** [comment]
@dandavison I'm sorry for the late response, but I just installed delta 0.16.5 and I confirm this is now working. Thank you!
